src/main/java/ch/njol/skript/expressions/ExprDefaultValue.java

```diff
@@ -0,0 +1,119 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.LiteralUtils;
+import ch.njol.skript.util.Utils;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.lang.reflect.Array;
+
+@Name("Default Value")
+@Description("A shorthand expression for giving things a default value. If the first thing isn't set, the second thing will be returned.")
+@Examples({"broadcast {score::%player's uuid%} otherwise \"%player% has no score!\""})
+@Since("INSERT VERSION")
+@SuppressWarnings("null")
+public class ExprDefaultValue<T> extends SimpleExpression<T> {
+
+    static {
+        Skript.registerExpression(ExprDefaultValue.class, Object.class, ExpressionType.COMBINED,
+                "%objects% (otherwise|?|or else) %objects%");
```

Comment:
How much did you test this? You know using objects in an expression messes up list variable literals so it could be harmful.

Reply:
very much however i don't know what you mean about literals

Reply:
```
set {_variable::*} to {_test} otherwise 1 and 5
```
gets parsed as:
```
set {_variable::*} to ({_test} otherwise 1 and 5)
```
instead of:
```
set {_variable::*} to ({_test} otherwise 1) and 5
```

Reply:
i don't really see the issue here, that sounds fine.

Reply:
I have to agree with Pika, I think if you want the last syntax you should have to include the parentheses. At least IMO the intuitive way for `set {_var::*} to {_test} otherwise 1 and 5` to be parsed is as `set {_var::*} to ({_test} otherwise 1 and 5)`, that's how the grouping reads by default logically. If you wanted the 5 separate, I feel like it'd only make sense that you have to add parentheses or put it first (`set {_var::*} to 5 and {_test} otherwise 1`)